lms/validation/__init__.py

```diff
@@ -81,8 +82,17 @@ def wrapper_view(context, request):
             # Use the view's configured schema to validate the request,
             # and make the validated and parsed request params available as
             # request.parsed_params.
+            schema = info.options["schema"]
+
+            # Schemas can just be dicts or they can be actual
+            # marshmallow.Schema subclasses, in which case we need to
+            # instantiate the class.
+            if isinstance(schema, marshmallow.Schema):
+                schema = schema()
+                schema.context["request"] = request
```

Comment:
Field validator methods in schema classes will be able to access the request as `self.context["request"]`. This enables a validator method for the OAuth 2 state param to not only check that state is present and is a string, but that it matches the saved state in `request.session`.